File: doc-examples/jdbc-example-records-java/src/main/java/example/CartItem.java
package example;

import io.micronaut.core.annotation.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;

@MappedEntity
public record CartItem(@Id @GeneratedValue @Nullable
                       Long id,
                       String name,
                       @Nullable
                       @Relation(value = Relation.Kind.MANY_TO_ONE)
                       Cart cart) {

    public CartItem(String name) {
        this(null, name, null);
    }

}

File: doc-examples/mongo-example-kotlin-ksp/src/main/kotlin/example/CartItem.kt
package example

import io.micronaut.core.annotation.Nullable
import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation
import org.bson.types.ObjectId

@MappedEntity
data class CartItem(
        @field:Id @GeneratedValue
        val id: ObjectId?,
        val name: String,
        @Nullable
        @Relation(value = Relation.Kind.MANY_TO_ONE)
        val cart: Cart?
) {
    constructor(name: String) : this(null, name, null)
}
